### DISCUSSION TITLE
Export as ctf file

### DISCUSSION THREAD

**Discussion by ArnaudPROIETTI** [mtex-toolbox/mtex#908] [Ask Anything]
Hello everyone,

I have an issue when I want to export my data as a ctf file. 

It appears when I want to filter small grains : e.g. : 
`ebsd(grains(grains.grainSize < 10)) = []` or `ebsd = ebsd(grains(grains.grainSize > 10))`

Removes fraction is not appearing in the resulting file and thus create an apparent distorsion :
![IPF_Z_5_6_1](https://user-images.githubusercontent.com/81309895/112322844-8990cb80-8cb1-11eb-9369-7de4c158825a.jpg)

This problem does not appears when I am using the previous version (5.4.0)

![IPF_Z_5_4_0](https://user-images.githubusercontent.com/81309895/112322923-9ca39b80-8cb1-11eb-8ccc-ae9d05de590d.jpg)

Thank you for your help

**Comment by kilir** [comment]
Hi,
could you post all the relevant lines? I can't reproduce your observation.
Cheers,
Rüdiger

**Comment by ArnaudPROIETTI** [comment]
Thanks for your quick answer. 

Here is the corresponding lines : 
`mtexdata forsterite`
`[grains,ebsd.grainId] = calcGrains(ebsd('indexed'), 'angle', 10*degree);`
`ebsd(grains(grains.grainSize < 10)) = []`
`ebsd.export('output.ctf')`

I didn't mention it but I work on a R2017b Matlab version.

Thanks again,

Arnaud

**Reply by kilir** [reply]
No problem here, R2020b, current Mtex
```
mtexdata forsterite
[grains,ebsd('indexed').grainId] = calcGrains(ebsd('indexed'), 'angle', 10*degree);
ebsd(grains(grains.grainSize < 10)) = []
plot(ebsd('f'),ebsd('f').orientations)
mtexTitle('before')
ebsd.export('output.ctf')
ebsd=EBSD.load('output.ctf')
nextAxis
plot(ebsd('f'),ebsd('f').orientations)
mtexTitle('after')
```
![something](https://user-images.githubusercontent.com/11697557/112343552-89013080-8cc3-11eb-929b-392c7e939843.png)

What are you plotting? 

Cheers,
Rüdiger

**Comment by ArnaudPROIETTI** [comment]
I tried exactly the same and I do not see any problem too. But when I look at the ctf file, I observed that lines which do not corresponds to defined criterion are deleted whereas with the 5.4.0 version line was remaining but all values was set to 0. I think that MTEX is reading (x,y) values instead of the other software which is just reading line number and grid size.

Thanks again for your help,

Cheers,
Arnaud

**Comment by ralfHielscher** [comment]
maybe you should do a `gridify` before exporting or just do not delete the
unwanted ebsd data but set them to notIndexed, i.e.,

ebsd(cond) = 'notIndexed'

Ralf.

>

**Comment by ArnaudPROIETTI** [comment]
Hello,

These two options are working perfectly !

Thanks a lot for your help,

Arnaud
